### ISSUE TITLE
programmatically prevent a package to be published that could clash with a stdlib module

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/nimble#576]
for context, see here: https://github.com/dom96/jester/issues/175#issuecomment-435199281

the check could be applied in a few places, eg:
* nimble publish (fail early without waiting for CI)
* in Nimble's CI, ie https://github.com/nim-lang/nimble/blob/master/.travis.yml (this'll guarantee it always works; eg in case a commit is somehow merged in alternative way)

eg:

`nimble publish` would fail for a module `strutils` (since it's in `Nim/lib/pure/strutils`) but not for a module called `varints` (since it's in `Nim/lib/std` as all new Nim modules)

* `nimble check` could also check for that

**Comment by dom96** [comment]
Nimble's CI? you mean the package repos CI right?

**Comment by timotheecour** [comment]
correct, I meant https://github.com/nim-lang/packages/blob/master/.travis.yml

so the check could be catch-all in https://github.com/nim-lang/packages/blob/master/.travis.yml

as well as catch-early in `nimble check` and `nimble publish`

**Comment by dom96** [comment]
Sure, we can do this. Now that 1.0 is close we can have a pretty stable list of stdlib module names too.
